src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/formulapredictionpeaklist/FormulaPredictionPeakListTask.java

```diff
@@ -190,13 +216,13 @@ public void run() {
 
   }
 
-  private void checkConstraints(IMolecularFormula cdkFormula, PeakListRow peakListRow) {
+  private ResultFormula checkConstraints(IMolecularFormula cdkFormula, PeakListRow peakListRow) {

```

Comment:
Think it now makes more sense for this function to return boolean instead of ResultFormula

Reply:
hm I think it is the easiest way to get the formula of the checked cdkFormula. The old void method added the ResultFormula to a list. I have to loop through that list as well if I change the return type to boolean. Maybe we should just rename the method, e.g." getCheckedFormula"? What do you think?

Reply:
okay, I have an other idea. Lets remove the "ResultingFormulas" List. It is not necessary, because only the formula is later used in the peaklist. All the other information the "ResultFormula" data type holds are not used.

Reply:
Yes, it seems the ResultingFormulas is redundant with your possibleFormulas.

Reply:
You can rename the checkConstraints() to getCheckedFormula()